test/SlowTests/Cluster/ClusterTransactionTests.cs

```diff
@@ -278,11 +278,14 @@ public async Task CanPreformSeveralClusterTransactions(int numberOfNodes)
                                 }
 
                                 if (numberOfNodes > 1)
-                                    await ActionWithLeader(l =>
+                                    await ActionWithLeader(async l =>
                                     {
+                                        using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
+                                        var waitForLeaderChangeTask = l.ServerStore.Engine.WaitForLeaderChange(cts.Token);
                                         l.ServerStore.Engine.CurrentLeader?.StepDown();
-                                        return Task.CompletedTask;
+                                        await waitForLeaderChangeTask;
```

Comment:
why would that matter in this test?

Reply:
I start a task that stops when the leader is changing, that's how I can promise that in the next iteration of the loop, a leader will be already chosen and then the exception will never be thrown.